dockerfile

```diff
@@ -0,0 +1,44 @@
+# Stage 1: Build the NestJS application
+FROM node:18-alpine AS builder
+
+# Set the working directory in the container
+WORKDIR /app
+
+# Copy package.json and package-lock.json for efficient layer caching
+COPY package*.json ./
+
+# Install dependencies
+RUN npm install
+
+# Copy the rest of the application code
+COPY . .
```

Comment:
instead of copying everything we should only copy what is required. This includes:
- `tsconfig.json`
- `tsconfig.build.json`
- `nest-cli.json`
- `webpack.config.js`
- `src`
- `config`
- `migrations`
- `package-lock.json`
- `package.json`

Reply:
Even though it's done in the `build` section, I'm pretty sure that it's the standard practice to just copy the required files.